sql/session.go

```diff
@@ -148,6 +146,13 @@ type Session interface {
 	SetPrivilegeSet(newPs PrivilegeSet, counter uint64)
 	// ValidateSession provides integrators a chance to do any custom validation of this session before any query is executed in it. For example, Dolt uses this hook to validate that the session's working set is valid.
 	ValidateSession(ctx *Context, dbName string) error
+	// SetTransactionDatabase is called when a transaction begins, and is set to the name of the database in scope for
+	// that transaction. GetTransactionDatabase can be called by integrators to retrieve this database later, when it's
+	// time to commit via TransactionSession.CommitTransaction. This supports implementations that can only support a
+	// single database being modified per transaction.
+	SetTransactionDatabase(dbName string)
```

Comment:
talked in person about how this is a workaround, and future work refactors a similar functionality into `TransactionSession` to allow for tracking before-transaction watermark metadata for in-scope databases. TODO might have been helpful but makes sense to me